File: frontend/src/types/team.ts
import type { Pokemon } from "./pokemon";

export interface Team {
  id: number;
  client_id: string;
  name: string;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface TeamPokemonSlot {
  slot: number;
  pokemon: Pokemon;
}

export interface TeamDetail extends Team {
  roster: TeamPokemonSlot[];
}

File: backend/app/models/team.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db import Base


class Team(Base):
    __tablename__ = "teams"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    client_id: Mapped[str] = mapped_column(String, index=True, nullable=False)
    name: Mapped[str] = mapped_column(String, nullable=False)

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False
    )

    roster: Mapped[list["TeamPokemon"]] = relationship(
        back_populates="team", cascade="all, delete-orphan", order_by="TeamPokemon.slot"
    )
